LayoutTests/imported/w3c/web-platform-tests/webcodecs/videoFrame-construction.crossOriginSource.sub.html

```diff
@@ -1,222 +1,219 @@
+      },
+      {
+        title: 'Test creating a VideoFrame with a cross-origin HTMLVideoElement',
+        factory: () => {
+          return new Promise(async (resolve, reject) => {
+            const video = document.createElement('video');
+            on_frame_available(video, () => resolve(video));
+            video.onerror = reject;
+            video.src = CROSSORIGIN_BASE + await getVideoFilename();
+          });
+        },
+        should_throw: true,
+      },
+      {
+        title: 'Test creating a VideoFrame with a CORS enabled cross-origin HTMLVideoElement without setting crossorigin',
+        factory: () => {
+          return new Promise(async (resolve, reject) => {
+            const video = document.createElement('video');
+            on_frame_available(video, () => resolve(video));
+            video.onerror = reject;
+            video.src = CROSSORIGIN_BASE + await getVideoFilename() + '?pipe=header(Access-Control-Allow-Origin,*)';
+          });
+        },
+        should_throw: true,
+      },
+      {
+        title: 'Test creating a VideoFrame with a CORS enabled cross-origin HTMLVideoElement with crossorigin="anonymous"',
+        factory: () => {
+          return new Promise(async (resolve, reject) => {
+            const video = document.createElement('video');
+            on_frame_available(video, () => resolve(video));
+            video.onerror = reject;
+            video.crossOrigin = 'anonymous';
+            video.src = CROSSORIGIN_BASE + await getVideoFilename() + '?pipe=header(Access-Control-Allow-Origin,*)';
+          });
+        },
+        should_throw: false,
+      },
+    ];
 
-TESTS.forEach(test => run_test(test));
+    TESTS.forEach(test => run_test(test));
 
-function run_test(test) {
-  promise_test(async t => {
-    const source = await test.factory();
-    if (test.should_throw) {
-      assert_throws_dom('SecurityError', () => { create_frame(source); });
-    } else {
-      create_frame(source);
+    function run_test(test) {
+      promise_test(async t => {
+        const source = await test.factory();
+        if (test.should_throw) {
+          assert_throws_dom('SecurityError', () => { create_frame(source); });
+        } else {
+          create_frame(source);
+        }
+      }, test.title);
     }
-  }, test.title);
-}
 
-function create_frame(img) {
-  let frame = new VideoFrame(img, {timestamp: 0});
-  frame.close();
-}
+    function create_frame(img) {
+      let frame = new VideoFrame(img, { timestamp: 0 });
+      frame.close();
+    }
 
-function on_frame_available(video, callback) {
-  if ('requestVideoFrameCallback' in video)
-    video.requestVideoFrameCallback(callback);
-  else
-    video.onloadeddata = callback;
-}
+    function on_frame_available(video, callback) {
+      if ('requestVideoFrameCallback' in video)
+        video.requestVideoFrameCallback(callback);
+      else
+        video.onloadeddata = callback;
+    }
 
-</script>
+  </script>
 </body>
+
```

Comment:
Are these changes needed?
It seems related to tabs/whitespaces and I do not see any -expected.txt change

Reply:
Yeah, I imported this test anew from upstream and without one of the changes I make here we would end up failing this test: https://wpt.fyi/results/webcodecs/videoFrame-construction.crossOriginSource.sub.html.

https://github.com/web-platform-tests/wpt/commit/9ff836132efe33400736a4c1fe0c76bd4e6808ab

This is the test that is being changed:

> Test creating a VideoFrame with a CORS enabled cross-origin SVGImageElement with crossorigin="anonymous"